src/rpc/README.txt

```diff
@@ -0,0 +1,60 @@
+--------------------------------------------------------------------------------
+Python -> OCaml RPC
+--------------------------------------------------------------------------------
+
+The mechanism implemented here and in `ocaml.py` in pysemgrep allows us to
+incrementally migrate functionality from Python to OCaml in support of our
+osemgrep effort. Instead of being limited to the one main call out to
+semgrep-core, we can now make additional calls from arbitrary points in the
+Python code.
+
+To add a function:
+
+- Update the `function_call` and `function_return` types in
+  `semgrep_output_v1.atd` in the `semgrep-interfaces` module.
+- Update `RPC.ml` to handle the function call and return the appropriate value.
+- Update `ocaml.py` to make the function call.
+- Call the function in `ocaml.py` from wherever you need to in the Python code.
+
+Suggested best practices:
+
+- Avoid exposing the ATD-generated types to clients of `ocaml.py`.
```

Comment:
not sure I agree with this one.

Reply:
This is the root of your comments above about preferring `out.$X` over `ocaml.$X`. I defined more ergonomic types in `ocaml.py` for clients to use, but it is a bit of duplication. The problem with the ATD generated types is they are kind of awkward to use in Pytyhon and you have to grab the `value` property. But it's not a big deal, and I'm happy to remove this and refactor accordingly.

Reply:
I agree they are a bit awkward, but there are many places in pysemgrep where we use them already, so I don't think it's a big problem if we use them also here.